### PR TITLE
Random Battle Move and Ability Updates

### PR DESCRIPTION
**Pull request by ascriptmaster**
Reject Mach Punch as the only STAB move.

Reject Bullet Punch as the only STAB move if the Pokemon does not have
Technician. Affects Metagross and Mega Lucario.

Reject Sheer Force with Fake Out in Doubles. Affects Hariyama.

Reject Flash Cannon with Iron Head in Doubles to match the rule in
Singles. Affects Aegislash.

---

I want to discuss Cherrim. In Singles, since it is so frail, I don't think it has as much of a use for the sun if it doesn't have Weather Ball/Solarbeam (although I guess the Special Defense boost helps). I was thinking of adding other moves like Dazzling Gleam or Healing Wish to compensate. Maybe Giga Drain could also be added but Cherrim's low stats means it really might want Energy Ball in randbats for the raw power (as opposed to it wanting Giga Drain in PU)

### CODE REVIEW

**Comment by TheImmortal** [comment]
Treat Mach Punch the same as Bullet Punch because Breloom.

Cherrim is pretty weak for Dazzling Gleam. Giga Drain would be better if it didn't have Synthesis but for now I'd keep it with Energy Ball. Growth is an option. I'm worried Healing Wish would get poor sets because the only viable set would be Energy Ball / HP Ice / Synthesis / Healing Wish. Anything else it should not be getting.

**Comment by TheImmortal** [comment]
We could treat them as separate Pokemon so Sunshine gets Sunny Day but the normal forme doesn't!

`["energyball","dazzlinggleam","hiddenpowerfire","synthesis","healingwish"]`
(Dazzling Gleam is okay here cause it'd get Life Orb; there's also no other move options...)

`["sunnyday","solarbeam","weatherball","hiddenpowerice","synthesis"]`

**Comment by ascriptmaster** [comment]
So something like this? (Taking the Smogon set mostly for the Singles set)

I made Cherrim's Doubles sets the same (with Protect in place of HP Ice) because Flower Gift can support its teammate as well.

**Comment by TheImmortal** [comment]
No need for HP Rock. It isn't even in usage stats.

**Comment by ascriptmaster** [comment]
OK. And now a new thing to help out Emolga to prevent it from having Charge Beam / Thunderbolt / Substitute / Baton Pass, which isn't a very good set.

**Comment by TheImmortal** [comment]
Well, it's a Baton Pass set and not a sweeping set.

**Comment by TheImmortal** [comment]
Perhaps rejecting Thunderbolt with Charge Beam if it has no other Special Attacks AND not counting Charge Beam as STAB. So it would get Charge Beam / Air Slash / Substitute / Baton Pass.

**Comment by ascriptmaster** [comment]
All right, done (kind of). I'm still rejecting Charge Beam with Thunderbolt if it has no other Special Attacks, and rejecting Thunderbolt with Charge Beam if it has at least one other Special Attack.

Also, no matter how much of a Baton Pass set it is, double STAB like that isn't very helpful

File: data/scripts.js [hunk 1]
```diff
@@ -1297,11 +1301,14 @@ exports.BattleScripts = {
...
 					break;
+				case 'chargebeam':
+					if (hasMove['thunderbolt'] && counter.Special < 3) rejected = true;
+					break;
 				case 'thunder':
 					if (hasMove['thunderbolt'] && !hasMove['raindance']) rejected = true;
 					break;
 				case 'thunderbolt':
-					if (hasMove['discharge'] || (hasMove['thunder'] && hasMove['raindance']) || (hasMove['voltswitch'] && hasMove['wildcharge'])) rejected = true;
+					if (hasMove['discharge'] || (hasMove['chargebeam'] && counter.Special > 2) || (hasMove['thunder'] && hasMove['raindance']) || (hasMove['voltswitch'] && hasMove['wildcharge'])) rejected = true;
```
Comment (TheImmortal) [inline data/scripts.js]: I don't feel good about this after generating some sets. This will greatly reduce the use of Thunderbolt.

Comment (TheImmortal) [inline data/scripts.js]: Just keep your check and remove this one. Keep Charge Beam in the `NoStab` list though.

**Comment by ascriptmaster** [comment]
Done. I also switched around some moves for Mega Glalie.

In both Singles and Doubles, Taunt is unheard of and Freeze-Dry is very common, so Taunt was removed and Freeze-Dry was added.

In Singles, Crunch was also very uncommon. After some discussion in the staff room, we believe that Crunch isn't that necessary when you have Freeze-Dry (hits Waters) and Earthquake (hits Steels and Fires, and also hits non-levitating Ice harder than Crunch) for Glalie. Do you think it's worth using, @TheImmortal?

File: data/scripts.js [hunk 2]
```diff
@@ -1366,7 +1376,7 @@ exports.BattleScripts = {
 				case 'bonemerang': case 'precipiceblades':
 					if (hasMove['earthquake']) rejected = true;
 					break;
-				case 'icebeam':
+				case 'icebeam': case 'iceshard':
 					if (hasMove['blizzard'] || hasMove['freezedry']) rejected = true;
```
Comment (TheImmortal) [inline data/scripts.js]: eh, add a separate case because this affects Abomasnow who is good with Ice Shard and Blizzard

**Comment by ascriptmaster** [comment]
Ice Shard is now a seperate case

**Comment by zJoshz** [comment]
I definitely don't think crunch is worth using.

**Pull request merged**